src/material/tabs/tab-nav-bar/tab-nav-bar.ts

```diff
@@ -65,97 +55,88 @@ const _MatTabNavMixinBase: CanDisableRippleCtor & CanColorCtor & typeof MatTabNa
-    implements AfterContentChecked, AfterContentInit, CanColor, CanDisableRipple, OnDestroy {
-
-  /** Subject that emits when the component has been destroyed. */
-  private readonly _onDestroy = new Subject<void>();
-
-  private _activeLinkChanged: boolean;
-  private _activeLinkElement: ElementRef<HTMLElement> | null;
-
-  @ViewChild(MatInkBar, {static: true}) _inkBar: MatInkBar;
+export class MatTabNav extends MatPaginatedTabHeader implements AfterContentChecked,
+  AfterContentInit, OnDestroy {
 
   /** Query list of all tab links of the tab navigation. */
-  @ContentChildren(forwardRef(() => MatTabLink), {descendants: true})
-  _tabLinks: QueryList<MatTabLink>;
+  @ContentChildren(forwardRef(() => MatTabLink), {descendants: true}) _items: QueryList<MatTabLink>;
+  @ViewChild(MatInkBar, {static: true}) _inkBar: MatInkBar;
+  @ViewChild('tabListContainer', {static: true}) _tabListContainer: ElementRef;
+  @ViewChild('tabList', {static: true}) _tabList: ElementRef;
+  @ViewChild('nextPaginator', {static: false}) _nextPaginator: ElementRef<HTMLElement>;
+  @ViewChild('previousPaginator', {static: false}) _previousPaginator: ElementRef<HTMLElement>;
 
   /** Background color of the tab nav. */
   @Input()
   get backgroundColor(): ThemePalette { return this._backgroundColor; }
   set backgroundColor(value: ThemePalette) {
-    const nativeElement: HTMLElement = this._elementRef.nativeElement;
-
-    nativeElement.classList.remove(`mat-background-${this.backgroundColor}`);
+    const classList = this._elementRef.nativeElement.classList;
+    classList.remove(`mat-background-${this.backgroundColor}`);
 
     if (value) {
-      nativeElement.classList.add(`mat-background-${value}`);
+      classList.add(`mat-background-${value}`);
     }
 
     this._backgroundColor = value;
   }
   private _backgroundColor: ThemePalette;
 
+  /** Whether the ripple effect is disabled or not. */
+  @Input()
+  get disableRipple() { return this._disableRipple; }
+  set disableRipple(value: any) { this._disableRipple = coerceBooleanProperty(value); }
+  private _disableRipple: boolean = false;
+
+  /** Theme color of the nav bar. */
+  @Input() color: ThemePalette = 'primary';
+
   constructor(elementRef: ElementRef,
-              @Optional() private _dir: Directionality,
-              private _ngZone: NgZone,
-              private _changeDetectorRef: ChangeDetectorRef,
-              private _viewportRuler: ViewportRuler) {
-    super(elementRef);
+              @Optional() dir: Directionality,
+              ngZone: NgZone,
+              changeDetectorRef: ChangeDetectorRef,
+              viewportRuler: ViewportRuler,
+              /**
+               * @deprecated @breaking-change 9.0.0 `platform` parameter to become required.
+               */
+              @Optional() platform?: Platform) {
+    super(elementRef, changeDetectorRef, viewportRuler, dir, ngZone, platform);
+  }
+
+  protected _itemSelected() {
```

Comment:
Could this be optional in the abstract class?

Reply:
Apparently optional abstract methods aren't supported by TS. I can mark it as optional, but it still requires it to be defined on the sub-class.